lib/origen_arm_debug/mem_ap.rb

```diff
@@ -100,6 +100,12 @@ def write_register(reg_or_val, options = {});
         addr = reg_or_val                 # if not a register, use the 'val' as target addr
         wdata = options[:wdata]
       end
+
+      if reg_or_val.is_a? Numeric  # if not a register or value, the calling application generates the addr and register data to use and passes it as options to the method.
+        addr = options[:reg_addr]
+        wdata = options[:reg_data]
```

Comment:
Ugh, just noticed above that in the case of supplying a value argument that is taken as the address. That is kind of bass ackwards compared to how other drivers work. The required argument is always the data, and if you are supplying a value (i.e. not an object with an address method), then you must also supply an :address option.